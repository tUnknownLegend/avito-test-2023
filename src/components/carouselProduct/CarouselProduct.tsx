import {Carousel} from 'antd';
import './carouselProduct.scss';

/**
 * Component to render carousel of products' photos
 */
function CarouselProduct({imgs} : {imgs: Array<string>}) {
    const carouselContent = imgs.map((img) => (
        <img className="product-images" src={img} alt={img} key={img}>
        </img>));

    return (
        <div className="product-img-container">
            <Carousel autoplay={true}>
                {carouselContent}
            </Carousel>
        </div>
    );
}

export default CarouselProduct;
