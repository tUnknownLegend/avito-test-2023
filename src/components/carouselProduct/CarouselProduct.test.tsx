import {render} from '@testing-library/react';
import CarouselProduct from './carouselProduct.tsx';

afterEach(jest.clearAllMocks);

const imgs = [
    'https://www.freetogame.com/g/181/thumbnail.jpg',
    'https://www.freetogame.com/g/181/8BitMMO-1.jpg',
];
describe('Carousel Product test', () => {
    it('should render correctly', () => {
        const rendered = render(<CarouselProduct imgs={imgs}/>);

        expect(rendered.asFragment()).toMatchSnapshot();
    });
});
