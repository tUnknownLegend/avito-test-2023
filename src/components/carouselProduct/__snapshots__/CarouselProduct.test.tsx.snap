// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Carousel Product test should render correctly 1`] = `
<DocumentFragment>
  <div
    class="product-img-container"
  >
    <div
      class="ant-carousel css-dev-only-do-not-override-17a39f8"
    >
      <div
        class="slick-slider slick-initialized"
        dir="ltr"
      >
        <div
          class="slick-list"
        >
          <div
            class="slick-track"
            style="opacity: 1; transform: translate3d(0px, 0px, 0px);"
          >
            <div
              aria-hidden="true"
              class="slick-slide slick-cloned"
              data-index="-1"
              style="width: 0px;"
              tabindex="-1"
            >
              <div>
                <img
                  alt="https://www.freetogame.com/g/181/8BitMMO-1.jpg"
                  class="product-images"
                  src="https://www.freetogame.com/g/181/8BitMMO-1.jpg"
                  style="width: 100%; display: inline-block;"
                  tabindex="-1"
                />
              </div>
            </div>
            <div
              aria-hidden="false"
              class="slick-slide slick-active slick-current"
              data-index="0"
              style="outline: none; width: 0px;"
              tabindex="-1"
            >
              <div>
                <img
                  alt="https://www.freetogame.com/g/181/thumbnail.jpg"
                  class="product-images"
                  src="https://www.freetogame.com/g/181/thumbnail.jpg"
                  style="width: 100%; display: inline-block;"
                  tabindex="-1"
                />
              </div>
            </div>
            <div
              aria-hidden="true"
              class="slick-slide"
              data-index="1"
              style="outline: none; width: 0px;"
              tabindex="-1"
            >
              <div>
                <img
                  alt="https://www.freetogame.com/g/181/8BitMMO-1.jpg"
                  class="product-images"
                  src="https://www.freetogame.com/g/181/8BitMMO-1.jpg"
                  style="width: 100%; display: inline-block;"
                  tabindex="-1"
                />
              </div>
            </div>
            <div
              aria-hidden="true"
              class="slick-slide slick-cloned"
              data-index="2"
              style="width: 0px;"
              tabindex="-1"
            >
              <div>
                <img
                  alt="https://www.freetogame.com/g/181/thumbnail.jpg"
                  class="product-images"
                  src="https://www.freetogame.com/g/181/thumbnail.jpg"
                  style="width: 100%; display: inline-block;"
                  tabindex="-1"
                />
              </div>
            </div>
            <div
              aria-hidden="true"
              class="slick-slide slick-cloned"
              data-index="3"
              style="width: 0px;"
              tabindex="-1"
            >
              <div>
                <img
                  alt="https://www.freetogame.com/g/181/8BitMMO-1.jpg"
                  class="product-images"
                  src="https://www.freetogame.com/g/181/8BitMMO-1.jpg"
                  style="width: 100%; display: inline-block;"
                  tabindex="-1"
                />
              </div>
            </div>
          </div>
        </div>
        <ul
          class="slick-dots slick-dots-bottom"
          style="display: block;"
        >
          <li
            class="slick-active"
          >
            <button>
              1
            </button>
          </li>
          <li
            class=""
          >
            <button>
              2
            </button>
          </li>
        </ul>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
